### PR TITLE
Table widget

### LINKED ISSUE #51: Document CollapsibleField
Describe how an object can be rendered as a collapsible sub form.

For example what does below do
```js
const uiSchema = {
    group1: {
      'ui:field': 'collapsible',
      'ui:collapsed': false
    }
  }
```
It renders collapsible sub form for `group1` parameter and is initially expanded.

### PR DESCRIPTION
**Pull request by sverhoeven**
In PR #50 we have several parameters which are array of objects. These can be rendered as a table in the form.

localhost_6006__path=_story_tablefield--array-of-objects-as-table

Table examples can be found in
* kitchensink catalog -> node5
* expert + guru catalog -> rigidbody -> symmetry or ad abnitio mode
* `yarn storybook` 

The usage of the table widget is document in the new `docs/ui/Schema.md` file. Also the other ui fields are documented in this file (fixes #51). 

Should be merged after #50 as this PR branches of from it.

### CODE REVIEW

**Comment by Peter9192** [comment]
@sverhoeven can you merge origin/main into this branch?

**Review by Peter9192 — APPROVED** [review]
Looks decent and seems to work well. Did I understand correctly that you already used this code elsewhere? Would it make sense to point to the source? Also, happy to see the extended documentation. Have you considered contributing some of these options back to react-json-schema-form? Would it make sense?

File: src/table/TableFieldTemplate.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  ) {
+    widths = props.uiSchema['ui:options'].widths as { [name: string]: string }
+  }
+  const headers = Object.entries(rowSchema).map(
+    ([key, s]: [string, any], i: number) => {
+      const title = required.has(key)
+        ? (
+          <label>
+            {s.title}
+            <span className='required'>*</span>
```
[the remark this answers is not in the document]
Comment (sverhoeven) [reply src/table/TableFieldTemplate.tsx:35]: Good catch, I have added a storybook and found that `*` was not shown everywhere it should have. Now it is shown in the column head and in popup.

File: src/table/TableFieldTemplate.css [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+.table-cell-field > div > .control-label,
+.table-cell-field > div > .field-description,
+.table-cell-field > .form-group > .form-group > .form-label,
+.table-cell-field > .form-group > .form-group > div > hr,
+.table-cell-field > .form-group > div > h5,  /* title of object */
+.table-cell-field > .form-group .mb-3, /* description of object */
+.table-cell-field .form-text /* description of object property */
+{
+    display: none;
+}
```
[the remark this answers is not in the document]
Comment (sverhoeven) [reply src/table/TableFieldTemplate.css]: added

**Comment by sverhoeven** [comment]
> Looks decent and seems to work well. Did I understand correctly that you already used this code elsewhere? Would it make sense to point to the source? Also, happy to see the extended documentation. Have you considered contributing some of these options back to react-json-schema-form? Would it make sense?

Thanks for reviewing.

I have added a comment to point to the source at https://github.com/iomega/paired-data-form/blob/master/app/src/fields/TableFieldTemplate.tsx 

I was thinking of making something similar to https://github.com/RXNT/react-json-schema-form-extras , but have not had the time yet

**Pull request merged**